src/charm.py

```diff
@@ -82,6 +83,21 @@ def _on_install(self, event: ops.InstallEvent) -> None:
         self.sshdebug.update_relation_data(host=str(self.state.ip_addr), fingerprints=fingerprints)
         self.unit.status = ops.ActiveStatus()
 
+    def _on_update_status(self, _: ops.UpdateStatusEvent) -> None:
+        """Check the health of the workload and restart if necessary."""
+        if not tmate.is_running():
+            logger.error("tmate-ssh-server is not running. Will restart.")
```

Comment:
Perhaps it is worth putting in some logs from the service into juju to help debug why the service might not be running?